docs/tutorial/workarounds.md

```diff
@@ -0,0 +1,204 @@
+    while len(remaining) > 1:
+        a = remaining.pop()
+        b = remaining.pop()
+        min_a_b = b - np.maximum(b - a, 0)
+        remaining.insert(0, min_a_b)
+    return remaining[0]
+
+inputset = [np.random.randint(0, 16, size=5) for _ in range(50)]
+circuit = fhe_min.compile(inputset)
+
+x1, x2, x3, x4, x5 = np.random.randint(0, 16, size=5)
+assert circuit.encrypt_run_decrypt([x1, x2, x3, x4, x5]) == min(x1, x2, x3, x4, x5)
+```
+
+## Retrieving a value within an encrypted array with an encrypted index
+
+This example show how to deal with an array and an encrypted index. It will create a "selection" array filled with `0` except for the requested index that will be `1`, and sum the products of all array values by this selection array:
+
+```python
+import numpy as np
+from concrete import fhe
+
+@fhe.compiler({"array": "encrypted", "index": "encrypted"})
+def indexed_value(array, index):
+    all_indices = np.arange(array.size)
+    index_selection = index == all_indices
+    selection_and_zeros = array * index_selection
+    selection = np.sum(selection_and_zeros)
+    return selection
+
+inputset = [(np.random.randint(0, 16, size=5), np.random.randint(0, 5)) for _ in range(50)]
+circuit = indexed_value.compile(inputset)
+
+array = np.random.randint(0, 16, size=5)
+
+index = np.random.randint(0, 5)
+assert circuit.encrypt_run_decrypt(array, index) == array[index]
+```
+
+## Filter an array with comparison (>)
+
+This example filters an encrypted array with an encrypted condition, here a `greater than` with an encrypted value.
+It packs all values with a selection bit, resulting from the comparison that allow the unpacking of only the filtered values:
+
+```python
+import numpy as np
+from concrete import fhe
+
+@fhe.compiler({"numbers": "encrypted", "threshold": "encrypted"})
+def filtering(numbers, threshold):
+    is_greater = numbers > threshold
+
+    shifted_numbers = numbers * 2  # open space for a single bit at the end
+    combined_numbers_and_is_greater = shifted_numbers + is_greater  # put is_greater to that bit
+
+    def extract(combination):
+        is_greater = (combination % 2) == 1  # extract is_greater back from packing
+        if_true = combination // 2  # if is greater is true, we unpack the number and use it
+        if_false = 0  # otherwise we set the element to zero
+        return np.where(is_greater, if_true, if_false)  # and apply the operation
+
+    return fhe.univariate(extract)(combined_numbers_and_is_greater)
+
+inputset = [(np.random.randint(0, 16, size=5), np.random.randint(0, 16)) for _ in range(50)]
+circuit = filtering.compile(inputset)
+
+numbers = np.random.randint(0, 16, size=5)
+threshold = np.random.randint(0, 16)
+assert np.array_equal(circuit.encrypt_run_decrypt(numbers, threshold), list(map(lambda x: x if x > threshold else 0, numbers)))
+
+```
+
+## Matrix Row/Col means
+
+In this example of Matrix operation, we are introducing a key concept when using `Concrete`:
```

Comment:
isn't it strange to have Concrete in `Concrete`?

Reply:
fixed

Reply:
tx